src/coreclr/src/jit/morph.cpp

```diff
@@ -2707,6 +2707,11 @@ void Compiler::fgInitArgInfo(GenTreeCall* call)
         indirectCellAddress->AsIntCon()->gtTargetHandle = (size_t)call->gtCallMethHnd;
 #endif
         indirectCellAddress->SetRegNum(REG_R2R_INDIRECT_PARAM);
+#ifdef TARGET_ARM
+        // Don't attempt to CSE this constant
+        // This hits an assert: Assertion failed 'candidates != candidateBit' in lsra.cpp Line: 3723
```

Comment:
could you please open an issue with a repro steps about that and place its number instead of line that could change?

Reply:
OK I will

Reply:
This constant has a specific register requirement, and I think that the same constant is used in another context with a different register requirement. So presumably the CSE value then has a conflict because it needs to be copied to both registers. I would probably restate this as:
"This constant has specific register requirements, and LSRA doesn't currently correctly handle them when the value is in a CSE'd local." or something to that effect.

Reply:
Yes, I think that that describe the issue, It only happens on Arm32, with physical register r4, we have a similar thing on Arm64 with a physical register r11 and that one works fortunately.